src/home_robot/home_robot/agent/goat_agent/goat_agent.py

```diff
@@ -0,0 +1,585 @@
+                    "blacklisted_targets_map": self.semantic_map.get_blacklisted_targets_map(
+                        e
+                    ),
+                    "timestep": self.total_timesteps[e],
+                }
+                for e in range(self.num_environments)
+            ]
+            if self.record_instance_ids:
+                for e in range(self.num_environments):
+                    vis_inputs[e]["instance_map"] = self.semantic_map.get_instance_map(
+                        e
+                    )
+        else:
+            vis_inputs = [{} for e in range(self.num_environments)]
+
+        return planner_inputs, vis_inputs
+
+    def reset_vectorized(self):
+        """Initialize agent state."""
+        self.total_timesteps = [0] * self.num_environments
+        self.sub_task_timesteps = [
+            [0] * self.max_num_sub_task_episodes
+        ] * self.num_environments
+        self.timesteps_before_goal_update = [0] * self.num_environments
+        self.last_poses = [np.zeros(3)] * self.num_environments
+        self.semantic_map.init_map_and_pose()
+        self.episode_panorama_start_steps = self.panorama_start_steps
+        self.reached_goal_panorama_rotate_steps = self.panorama_rotate_steps
+
+        self.landmark_found = False
+        self.seen_landmarks = []
+        self.reject_visited_targets = False
+        self.blacklist_target = False
+        self.current_task_idx = 0
+
+        if self.imagenav_visualizer is not None:
+            self.imagenav_visualizer.reset()
+        self.imagenav_obs_preprocessor.reset()
+
+        self.current_task_idx = 0
+
+        self.found_goal[:] = False
+        self.goal_map[:] *= 0
+        self.planner.reset()
+
+    def reset_vectorized_for_env(self, e: int):
+        """Initialize agent state for a specific environment."""
+        self.total_timesteps[e] = 0
+        self.sub_task_timesteps[e] = [0] * self.max_num_sub_task_episodes
+        self.timesteps_before_goal_update[e] = 0
+        self.last_poses[e] = np.zeros(3)
+        self.semantic_map.init_map_and_pose_for_env(e)
+        self.episode_panorama_start_steps = self.panorama_start_steps
+        self.reached_goal_panorama_rotate_steps = self.panorama_rotate_steps
+
+        self.landmark_found = False
+        self.seen_landmarks = []
+        self.reject_visited_targets = False
+        self.blacklist_target = False
+
+        self.current_task_idx = 0
+        self.planner.reset()
+
+    # ---------------------------------------------------------------------
+    # Inference methods to interact with the robot or a single un-vectorized
+    # simulation environment
+    # ---------------------------------------------------------------------
+
+    def reset(self):
+        """Initialize agent state."""
+        self.reset_vectorized()
+        self.planner.reset()
+
+    def act(self, obs: Observations) -> Tuple[DiscreteNavigationAction, Dict[str, Any]]:
+        """Act end-to-end."""
```

Comment:
The GOAT agent should map long-form English instructions to actions. It should infer the task type from these instructions instead of taking the task type as input.